packages/common/tests/collection-config-store.js

```diff
@@ -29,145 +32,171 @@ test.afterEach((t) =>
-    Body: JSON.stringify(t.context.collectionConfig)
+    Bucket: bucket,
+    Key: collectionConfigKey(dataType, dataVersion),
+    Body: JSON.stringify(collectionConfig)
   }).promise();
 
-  const collectionConfigStore = new CollectionConfigStore(t.context.bucket, t.context.stackName);
-  const fetchedCollectionConfig = await collectionConfigStore.get(
-    t.context.dataType,
-    t.context.dataVersion
-  );
+  const collectionConfigStore = new CollectionConfigStore(bucket, stackName);
+  const fetchedCollectionConfig = await collectionConfigStore.get(dataType,
+    dataVersion);
 
-  t.deepEqual(fetchedCollectionConfig, t.context.collectionConfig);
+  t.deepEqual(fetchedCollectionConfig, collectionConfig);
 });
 
 test.serial('get() does not hit S3 for a cached collection config', async (t) => {
+  const {
+    bucket,
+    collectionConfig,
+    collectionConfigKey,
+    dataType,
+    dataVersion,
+    stackName
+  } = t.context;
+
   await s3().putObject({
-    Bucket: t.context.bucket,
-    Key: t.context.collectionConfigKey(t.context.dataType, t.context.dataVersion),
-    Body: JSON.stringify(t.context.collectionConfig)
+    Bucket: bucket,
+    Key: collectionConfigKey(dataType, dataVersion),
+    Body: JSON.stringify(collectionConfig)
   }).promise();
 
-  const collectionConfigStore = new CollectionConfigStore(t.context.bucket, t.context.stackName);
+  const collectionConfigStore = new CollectionConfigStore(bucket, stackName);
 
   // Fetch the collection config once so it's in the cache
-  await collectionConfigStore.get(t.context.dataType, t.context.dataVersion);
+  await collectionConfigStore.get(dataType, dataVersion);
 
   // Delete the S3 bucket so the config can't be fetched from S3
-  await recursivelyDeleteS3Bucket(t.context.bucket);
+  await recursivelyDeleteS3Bucket(bucket);
 
   // This get() should use the cache
-  const fetchedCollectionConfig = await collectionConfigStore.get(
-    t.context.dataType,
-    t.context.dataVersion
-  );
+  const fetchedCollectionConfig = await collectionConfigStore.get(dataType,
+    dataVersion);
 
-  t.deepEqual(fetchedCollectionConfig, t.context.collectionConfig);
+  t.deepEqual(fetchedCollectionConfig, collectionConfig);
 });
 
 test.serial('get() throws an exception if the collection config could not be found', async (t) => {
+  const { bucket, dataVersion, stackName } = t.context;
   const invalidDataType = randomString();
-  const collectionConfigStore = new CollectionConfigStore(t.context.bucket, t.context.stackName);
-
-  try {
-    await collectionConfigStore.get(invalidDataType, t.context.dataVersion);
-    t.fail('Expected an error to be thrown');
-  } catch (err) {
-    t.is(err.message, `A collection config for data type "${invalidDataType}__${t.context.dataVersion}" was not found.`);
-  }
+  const collectionConfigStore = new CollectionConfigStore(bucket, stackName);
+  const collectionId = constructCollectionId(invalidDataType, dataVersion);
+
+  await t.throwsAsync(
```

Comment:
Ideally we should never look for a specific error message, but rather a specific error _type_. Since the type thrown is simply `Error`, I opted for making this _less_ fragile than before because now we're only looking for a value within the error message rather than looking for an exact match of the entire string.

I agree that this is still fragile, but I didn't want the refactoring to balloon into a larger job with this ticket. My recommendation would be to change the type of error thrown to some specific error type (not simply `Error`), then change this test to confirm the expected error type is thrown, without regard for the error message itself. Perhaps we should create a new ticket for such a refactoring (including determining an appropriate custom error type to create).

Reply:
Totally agreed that custom Errors are the right thing to do if you care about differentiation in this case.      

I think really this was more about test expressiveness (I shouldn't have used fragility), the test implies the error message is the thing that should be different, and that's not right.           Sounds like we're in agreement there, I'm fine punting that to a future refactor given it's a neutral change.